### ISSUE TITLE
Solid-icons "doesn't appear to be written in CJS, but also doesn't appear to be a valid ES module"

### ISSUE THREAD

**Issue by Chrysippus** [x64Bits/solid-icons#7]
Using Solid-Start to bootstrap a SolidJS project, I get the following error,
`TypeError: __vite_ssr_import_0__.template is not a function
    at eval (/node_modules/.pnpm/solid-icons@0.4.13_solid-js@1.4.3/node_modules/solid-icons/esm/IconWrapper.js:6:38)`

If I remove the icons but leave the package in package.json, the build log provides this clarification:
`solid-icons doesn't appear to be written in CJS, but also doesn't appear to be a valid ES module (i.e. it doesn't have "type": "module" or an .mjs extension for the entry point). Please contact the package author to fix.`

Basically, it means the icon imports don't work with Solid-Start.

So, this could probably use a fix! Otherwise, it means grabbing the raw SVGs and creating custom wrappers for them, which is tedious.

**Comment by x64Bits** [comment]
That's right, I'm already working on submitting the Cjs compatible package, thank you very much for reporting it.

**Comment by GodBleak** [comment]
Yeah, +1 for SSR support. When using any icons with SSR, it produces `TypeError: web.template is not a function` (from IconWrapper). Adding the solid-icons imports to `external` in the rollup config changes the error to one about using import outside a module.

**Comment by x64Bits** [comment]
I'm already working on SSR support, thank you very much for the suggestions, it seems to me that sending JSX can solve it, it worked when I tried, now I'm adapting the library to allow it.

**Comment by x64Bits** [comment]
Hello everyone, after a long work from version [0.5.0](https://www.npmjs.com/package/solid-icons/v/0.5.0) solid-icons now works with solid-start and SSR, I would like you to test what you think of the result, even though the JSX file is included in the lib, it's not being used as a wrapper, I haven't found a way to make it work with JSX yet.

If you come up with a way to use JSX and it turns out that it's better than sending the compile, you could correct it for the next version or if you like to send a PR.

**Comment by x64Bits** [comment]
The solid-start compilation is not working, after reviewing the trace it seems to me that the key "type": "module" is making the server side version look for the ESM version when compiling, if you have any ideas from solid-icons to be able to solve this, I will still be looking for a way to solve it on my part.

**Comment by Chrysippus** [comment]
I get the following error now on "pnpm build" with just a single solid-icon tossed on the index page:

`Error: 'template' is not exported by node_modules/.pnpm/solid-js@1.4.6/node_modules/solid-js/web/dist/server.js, imported by node_modules/.pnpm/solid-icons@0.5.0_solid-js@1.4.6/node_modules/solid-icons/lib/browser/index.module.js`

It really just throws that error for all of the imports in browser/index.module.js.

I'm not entirely sure what is going on here. I'm not familiar enough with rollup to know exactly how to fix this, but this is the error code I get along with the above error:

`code: 'MISSING_EXPORT',
  url: 'https://rollupjs.org/guide/en/#error-name-is-not-exported-by-module',`

**Comment by sya-ri** [comment]
> I get the following error now on pnpm build with just a single solid-icon tossed on the index page:
> 
> `Error: 'template' is not exported by node_modules/.pnpm/solid-js@1.4.6/node_modules/solid-js/web/dist/server.js, imported by node_modules/.pnpm/solid-icons@0.5.0_solid-js@1.4.6/node_modules/solid-icons/lib/browser/index.module.js
`

It occurs because of conditional imports. As an issue similar to this, [this](https://github.com/solidjs/solid/issues/335) is an issue that has already been resolved. [The solution](https://github.com/steelbrain/linter-ui-default/pull/629/files) is to set an alias path from `solid-js/web` to `solid-js/web/dist/web.js`. The library uses parcel instead of rollup, but it will be helpful. 

<img width="988" alt="image" src="https://user-images.githubusercontent.com/34268371/179357126-ac07efe9-744f-4d9a-928b-cd1caabe5b0e.png">

Of course, you can manually change the import of `node_modules/solid-icons/lib/browser/index.module.js` to build without error, but it isn't a solution.

Anyway, I like solid-js and solid-icons. I'm waiting for this issue to be resolved (Until then, icons are goodbye) :wink:

**Comment by x64Bits** [comment]
Hello @sya-ri thank you very much for exploring a possible solution.

What worked for me was to approach it from Vite's side and add exports path for each package in package.json according to import type, the problem occurs because by default Vite tries to resolve the package's .module file by importing in static generation and SSR, when in SSR mode it should go through the CJS that is by default exported in the package.json of the folder I don't know why this happens.

What they call conditional import, the only problem I observed with doing it this way is that if we want to expose something else from the library, such as the Wrapper, we must set it in the main package.json.

This week I'll be delivering version 1.0.0, I've rewritten the library and added a new icon pack, only I've also been working on the icon explorer website.

I am very grateful for all those who wrote helping to solve this, I will hope that you can test it as soon as I send it, I will leave a message in this thread to know if we can close it.

**Comment by sya-ri** [comment]
Dear @x64Bits,

Thank you for your excellent work. solid-icons 1.0.0 is working fine for my project using solid-start(SSR) ❤️

**Comment by x64Bits** [comment]
Hi everyone, i couldn't tell you during the day that I had sent version 1.0.0 of the source code and the library in npm, I hope you can try it, greetings!

Thank you very much @sya-ri.

your support guys has been very helpful, thank you!
